### PR TITLE
fix(explicitRollback): Add configurable timeout for serverGroup lookup from Clouddriver API

### PR DESCRIPTION
**Pull request by christosarvanitis**
When a user performs an Ad-Hoc Rollback through Deck, Orca is expecting to receive a response from Clouddriver about the serverGroup details in a hardcoded 5 seconds timeout. 
Although in most cases this is enough, in higher scale environments with thousands of AWS/ECS accounts and especially in applications that have a large amount of serverGroups the 5 sec timeout is too strict resulting in error during the serverGroup lookup.
TimeoutFailure

With this change we can configure the timeout (in seconds) through Orca config - while defaulting to the previous timeout of 5secs.

### CODE REVIEW

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/rollback/ExplicitRollback.groovy [hunk 1]
```diff
@@ -49,6 +50,8 @@ class ExplicitRollback implements Rollback {
   Boolean disableOnly
   Boolean enableAndDisableOnly
 
+  @Value('${rollback.explicitRollback.timeout:5}') Integer rollbackTimeout
```
[the remark this answers is not in the document]
Comment (jasonmcintosh) [reply orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/rollback/ExplicitRollback.groovy:53]: Agree with this!!!

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/rollback/ExplicitRollback.groovy [hunk 2]
```diff
@@ -163,11 +166,11 @@ class ExplicitRollback implements Rollback {
       })
 
       executor.submit(authenticatedRequest)
-        .get(5, TimeUnit.SECONDS)
+        .get(rollbackTimeout, TimeUnit.SECONDS)
         .get()  // not sure what would cause the Optional to not be present but we would catch and log it
     } catch(Exception e) {
       log.error('Could not generate resize stage because there was an error looking up {}', serverGroupName, e)
-      throw new SpinnakerException("failed to look up ${serverGroupName}", e)
+      throw new SpinnakerException("Failed Clouddriver look up for ${serverGroupName} in ${rollbackTimeout} sec. Consider increasing rollback.explicitRollback.timeout in orca profile.", e)
```
Comment (jasonmcintosh) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/rollback/ExplicitRollback.groovy:173]: Good log message here.

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/rollback/ExplicitRollback.groovy (hunk 1 above)
[the remark this answers is not in the document]
Comment (christosarvanitis) [reply orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/rollback/ExplicitRollback.groovy]: Sure! Just added the properties class. One note: the PR https://github.com/spinnaker/orca/pull/4383/files added a property:
```
rollback:
  timeout:
     enabled: true
```
With this change proposing to deprecate that in the next release and change it to:
```
rollback:
  dynamicRollback:
    enabled: false/true
  explicitRollback:
    timeout: 5
```
For now the previous property will work as well for the dynamicRollback feature flag

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/rollback/ExplicitRollback.groovy [hunk 3]
```diff
@@ -49,6 +51,9 @@ class ExplicitRollback implements Rollback {
   Boolean disableOnly
   Boolean enableAndDisableOnly
 
+  @Autowired
```
Comment (ovidiupopa07) [inline orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/rollback/ExplicitRollback.groovy:54]: I guess we should add the `@JsonIgnore` too

**Review by ovidiupopa07 — CHANGES_REQUESTED** [review]

File: orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/rollback/ExplicitRollback.groovy (hunk 3 above)
Comment (christosarvanitis) [reply orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/rollback/ExplicitRollback.groovy:54]: Done

**Review by ovidiupopa07 — APPROVED** [review]
LGTM 🚀

**Review by jasonmcintosh — APPROVED** [review]

**Comment by jasonmcintosh** [comment]
@mergifyi backport release-1.32.x release-1.33.x release 1-3.1.x

**Comment by jasonmcintosh** [comment]
@merigfyio  backport  release 1-31.x

**Comment by jasonmcintosh** [comment]
https://github.com/Mergifyio backport release-1.31.x

**Pull request merged**